code/game/objects/structures/tables_racks.dm

```diff
@@ -167,6 +167,9 @@
 	if(istype(I, /obj/item/storage/bag/tray))
 		var/obj/item/storage/bag/tray/T = I
 		if(T.contents.len > 0) // If the tray isn't empty
+			for(var/x in T.contents)
+				var/obj/item/item = x
+				AfterPutItemOnTable(item, user)

```

Comment:
this should have been a signal

Reply:
I'll keep that in mind for a follow-up PR, thank you.